specutils/fitting/fitmodels.py

```diff
@@ -1,20 +1,214 @@
+    'Lorentzian1D': {
+        'amplitude': None,
+        'x_0': None,
+        'fwhm':None 
+    },
+    'Voigt1D': {
+        'x_0': None,
+        'amplitude_L': None,
+        'fwhm_L': None,
+        'fwhm_G': None 
+    },
+    'Const1D': {
+        'amplitude': None 
+    }
+}
+
+def _set_parameter_estimators(model):
+    if model.__class__.__name__ in parameter_estimators:
+        model._constraints['parameter_estimator'] = parameter_estimators[model.__class__.__name__]
+    return model
+
+def estimate_parameters(spectrum, model):
+
+    if not 'parameter_estimator' in model._constraints:
+        model = _set_parameter_estimators(model)
+
+    # Estimate the parameters based on the estimators already
+    # attached to the model
+    if 'parameter_estimator' in model._constraints:
+        for param, estimator in model._constraints['parameter_estimator'].items():
+            setattr(model, param, estimator(spectrum))
+
+    # No estimators.
+    else:
+        raise Exception('No method to estimate parameters')
+
+    return model
+
+def _consecutive(data, stepsize=1):
+    return np.split(data, np.where(np.diff(data) != stepsize)[0]+1)
+
+
+def find_lines_threshold(spectrum, sigma=3):
+    """
+    Find the emission and absorption lines in a spectrum. The method
+    here is a very simple threshold based on the standard deviation
+    of the noise.
+
+    Parameters
+    ----------
+    spectrum : Spectrum1D
+        The spectrum object over which the equivalent width will be calculated.
+
+    sigma : float
+        Threshold on line finding.
+
+    Returns
+    -------
+    qtable: `~astropy.table.QTable`
+        Table of emission and absorption lines. Spectral axis values, line type and indices stored.
+    """
+
+    #
+    # Find the emission lines
+    #
+
+    inds = np.where(np.abs(spectrum.flux) > sigma*spectrum.flux.unit)[0]
+    pos_inds = inds[spectrum.flux.value[inds] > 0]
+    line_inds_grouped = _consecutive(pos_inds, stepsize=1)
+
+    if len(line_inds_grouped[0]) > 0:
+        emission_inds = [inds[np.argmax(spectrum.flux.value[inds])] for inds in line_inds_grouped]
+    else:
+        emission_inds = []
+
+    #
+    # Find the absorption lines
+    #
+
+    neg_inds = inds[spectrum.flux.value[inds] < 0]
+    line_inds_grouped = _consecutive(neg_inds, stepsize=1)
+
+    if len(line_inds_grouped[0]) > 0:
+        absorption_inds = [inds[np.argmin(spectrum.flux.value[inds])] for inds in line_inds_grouped]
+    else:
+        absorption_inds = []
+
+    #
+    # Create the QTable to return the lines
+    #
+
+    qtable = QTable()
+    qtable['lines'] = list(itertools.chain(*[spectrum.spectral_axis.value[emission_inds],
+                                             spectrum.spectral_axis.value[absorption_inds]]))*spectrum.spectral_axis.unit
+    qtable['line_type'] = ['emission']*len(emission_inds) + ['absorption']*len(absorption_inds)
+    qtable['index'] = list(itertools.chain(*[emission_inds, absorption_inds]))
```

Comment:
perhaps the units here should be `u.pixel`?

Reply:
I would suggest that it should be unitless. The primary use case would be for doing indexing into a numpy array or list so probably easiest to just leave it as a list of numbers. If you feel strongly otherwise I can change it.